packages/compiler-sfc/src/style/pluginScoped.ts

```diff
@@ -225,7 +225,7 @@ function rewriteSelector(
       (n.type !== 'pseudo' && n.type !== 'combinator') ||
       (n.type === 'pseudo' &&
         (n.value === ':is' || n.value === ':where') &&
-        !node)
+        (!node || !n.nodes.some(n => n.nodes.some(x => x.type === 'pseudo'))))
```

Comment:
This will cause pattern pollution and scoped will not work

Reply:
<img width="711" alt="image" src="https://github.com/user-attachments/assets/8b612cdb-9759-4e54-bf86-710f18136f6e">
<img width="696" alt="image" src="https://github.com/user-attachments/assets/2579edd8-8d32-4626-83b8-7a5eba500065">